### PR TITLE
London - Class 7 - Tanika - Javascript Core 2 - Week 1

### PR DESCRIPTION
**Pull request by tanikaDev**
completed mandatory exercises

### CODE REVIEW

File: mandatory/6-reading-list.js [hunk 1]
```diff
@@ -23,4 +23,39 @@ and if not, log a string like 'You still need to read "The Lord of the Rings" by
 
 **/
 
-let books = [];
+const { read } = require("fs");
```
Comment (louisechow) [inline mandatory/6-reading-list.js:26]: Not sure how line 26 got added, but you don't need this line.

File: mandatory/6-reading-list.js [hunk 2]
```diff
@@ -23,4 +23,39 @@ and if not, log a string like 'You still need to read "The Lord of the Rings" by
...
+
+let books = [{
```
Comment (louisechow) [inline mandatory/6-reading-list.js:28]: Great job defining the array and the objects inside it.

For the formatting of your array, I think this method is more common
```
let books = [
    {
        title: 'Harry Potter and the Deathly Hallows',
        author: 'J.K Rowling',
        read: true
    },
    {
        title: 'Think and Grow Rich',
        author: 'Napoleon Hill',
        read: false
    },
    //etc
]    
```
That way you can match indentation to see where the object starts and ends.

File: mandatory/6-reading-list.js [hunk 3]
```diff
@@ -23,4 +23,39 @@ and if not, log a string like 'You still need to read "The Lord of the Rings" by
...
+    read: true
+}, {
+    title: 'The Seat of the Soul',
+    author: 'Gary Zukav',
+    read: false
+}
+
+];
+
+for (let i=0; i<books.length; i++){
```
Comment (louisechow) [inline mandatory/6-reading-list.js:52]: Good solution, it works!

As a challenge, can you think of a way you could do this using one of the array methods you've learnt?

File: mandatory/3-groceries.js [hunk 4]
```diff
@@ -31,6 +31,19 @@ Exercise 1:
...
+  
+  weeklyGroceriesToBuy = weeklyGroceriesToBuy.concat(Object.values(weeklyMealPlan)[i]);
+};
+
+weeklyGroceriesToBuy = weeklyGroceriesToBuy.filter((word,index) => {
+    if (index === weeklyGroceriesToBuy.lastIndexOf(word))
+    return word;
+  } );
+
+console.log(weeklyGroceriesToBuy.sort())
```
Comment (mrehmm001) [inline mandatory/3-groceries.js:45]: Nice use of filter method to achieve this. Also, nice use of the lastIndeOf() method, but one thing I could also mention is, you can also use the .includes() method to check if an item is already is in the weeklyGroceriesToBuy array.

File: mandatory/3-groceries.js [hunk 5]
```diff
@@ -39,6 +52,13 @@ Exercise 2:
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
 
+for (let i=5; i < Object.keys(weeklyMealPlan).length; i++) {
+  
+  weekendGroceriesToBuy = weekendGroceriesToBuy.concat(Object.values(weeklyMealPlan)[i]);
+};
+
+console.log(weekendGroceriesToBuy);
```
Comment (mrehmm001) [inline mandatory/3-groceries.js:60]: Nicely done for this question, you clearly have shown good use of Object.keys and Object.values, and you started off from index 5 which is smart, now imagine if the weeklyMealPlan array was jumbled up, you would now need to check the entire array to find the corresponding arrays for Saturday and Sunday, how would you come about achieving this?

**Pull request closed without merging**